docs/pages/blog/masonry-post.md

```diff
@@ -0,0 +1,141 @@
+
+### `<MasonryItem />`
+
+This component is a CSS grid item that wraps around one and only child. Through the use of `React.Children.only()`, we enforce that only one child is passed to the component. MUI users can always create more `<MasonryItem />` to add more children.
+
+`<MasonryItem />` receives two special props: `columnSpan` and `defaultHeight`. `columnSpan` is the number of columns taken up by the component, while `defaultHeight`, which is provided for optional server-side rendering, is the initial height of the component. Both props receive a fixed value. Further details can be found in the [API documentation](https://mui.com/api/masonry-item/).
+
+`<MasonryItem />` utilizes a Web API called [ResizeObserver API](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver). It detects changes to the dimensions of an element. We use this API to observe the child passed to `<MasonryItem />`. Every time the child's height changes, which is commonly caused by window resizing or zooming, these two properties, `grid-row-end` and `padding-bottom`, also get updated. `grid-row-end` changes so that the height of `<MasonryItem />`, by spanning a correct number of rows, matches its child's height. `padding-bottom` changes so that the vertical gap between `<MasonryItem />`s stays correct.
+
+If the developer wants to use the column spanning feature and so passes the `columnSpan` prop to the component, the property `grid-column-end` is assigned the given value. For example, if `columnSpan` is `3`, `grid-column-end` is set to `span 3` and the component will span three columns.
+
+If the developer wants to use server-side rendering and so passes the `defaultHeight` prop to the component, the ResizeObserver API is not used so that the component, `<MasonryItem />`, is set to the given height and retains it. However, if `defaultHeight` is larger than the height of its child, there will be an unwanted gap between the component and the child. If it is less, some parts of the child will not be displayed. Hence, the prop `defaultHeight` should be used with special attention to enjoy the benefits of both a solid masonry and server-side rendering.
+
+## Yet...Chrome hit us from the back
+
+After finishing up the implementation, we found that Chrome limits the number of rows of each CSS grid to 1,000 at the maximum. Items overflowing beyond the 1,000th row are simply squashed. You can find a thread regarding this [here](https://bugs.chromium.org/p/chromium/issues/detail?id=688640).
+
+This is a serious problem for us because our implementation has the row height of `1px` and it is not that difficult for a masonry to go beyond a height of `1,000px`. As a hot fix, we set the row height to `2px`. So, your MUI Masonry will fail to be larger than `2,000px` on Chrome. It has no problem on Firefox or Safari, but since Chrome has the biggest market share of browsers, this definitely needs to be fixed.
```

Comment:
```suggestion
This is a serious problem for us because our implementation has the row height of `1px` and it is not that difficult for a masonry to go beyond a height of `1,000px`. As a hot fix, we set the row height to `2px`. So, your Masonry will fail to be larger than `2,000px` on Chrome. It has no problem on Firefox or Safari, but since Chrome has the biggest market share of browsers, this definitely needs to be fixed.
```